src/python/pants/util/dirutil.py

```diff
@@ -83,6 +83,15 @@ def safe_mkdir_for(path, clean=False):
   safe_mkdir(os.path.dirname(path), clean=clean)
 
 
+def safe_mkdir_for_all(paths):
```

Comment:
It would be a good idea to document why you might want to use this (E.g., when you'll be creating dirs for many files that are overwhelmingly likely to be in a small handful of dirs). I.e., why should I call this and not just call `safe_mkdir_for` on each path?

Reply:
Done